config/bootstrap.php

```diff
@@ -107,6 +107,8 @@
 if (Configure::read('debug')) {
     Configure::write('Cache._cake_model_.duration', '+2 minutes');
     Configure::write('Cache._cake_core_.duration', '+2 minutes');
+    // disable router cache during development
+    Configure::write('Cache._cake_routes_.duration', '+0 minutes');
```

Comment:
Won't this still unnecessarily write to cache only to be invalidated? A better option could be changing the middleware setup in `Application.php` to
`->add(new RoutingMiddleware($this, Configure::read('debug') ? null : '_cake_routes'));`

Reply:
@ADmad there is a problem 
when you are in product mode cache uses 
when you switch to debug mode cache disabled , without clearing 
but when you switch again to product mode , cake uses old cache

Reply:
Yes, also I think it's better to be consistent and write to cache in debug mode too to help the developer to deal with cache configuration.